File: proxy-server/Server.Domain/Shared/Response.cs
namespace Server.Domain.Shared;

public class Response
{
    public bool Success { get; set; } = true;

    public string? Message { get; set; }

    public string? ErrorMessage { get; set; }
}

public class Response<T> : Response where T : class
{
    public T? Data { get; set; }
} 

File: blip-cli/http/src/types/response.rs
use serde::Deserialize;
use serde_json::Value;

#[derive(Debug, Deserialize)]
pub struct ProxyResponse {
    #[serde(rename = "success")]
    pub sucesss: bool,

    #[serde(rename = "data")]
    pub data: Value,

    #[serde(rename = "message")]
    pub message: Option<String>,

    #[serde(rename = "errorMessage")]
    pub error_message: Option<String>
}